### PR TITLE
added director userInfo to response and check of userData

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

File: src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs [hunk 1]
```diff
@@ -71,30 +71,41 @@ private List<UserData> GetUsers(List<Guid> usersIds, List<string> errors)
...
-
             foreach (var department in dbDepartments)
             {
-                UserInfo director = null;
-                List<UserData> usersData = GetUsers(department.Users.Select(x => x.UserId).ToList(), errors);
+                List<Guid> usersIds = department.Users.Select(x => x.UserId).ToList();
+
+                if (department.DirectorUserId != null)
+                {
+                    usersIds.Add((Guid)department.DirectorUserId);
```
Comment (DarkFelas) [inline src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs:84]: department.DirectUserId.Value;

Comment (DarkFelas) [reply src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs:84]: and before add directorID plz check if list already contains this ID

File: src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs [hunk 2]
```diff
@@ -71,30 +71,41 @@ private List<UserData> GetUsers(List<Guid> usersIds, List<string> errors)
...
                 if (department.DirectorUserId != null && usersData.Any())
                 {
-                    director = _userMapper.Map(usersData.FirstOrDefault(x => x.Id == department.DirectorUserId));
+                    var directorUserData = usersData.FirstOrDefault(x => x.Id == department.DirectorUserId);
+                    director = directorUserData == null ? null : _userMapper.Map(directorUserData);
                 }
 
-                departments.Add(_departmentMapper.Map(department, director, usersData.Select( _userMapper.Map).ToList()));
+                List<UserInfo> usersInfo = new();
+                foreach(UserData userData in usersData)
```
Comment (DarkFelas) [inline src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs:97]: foreach(UserData userData in usersData.Where(us => us != null))

File: src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs (hunk 1 above)
Comment (MariyaSin) [reply src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs]: done

File: src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs (hunk 2 above)
Comment (MariyaSin) [reply src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs]: done

**Pull request merged**